exla/test/exla/random_test.exs

```diff
@@ -39,4 +39,23 @@ defmodule EXLA.NxRandomTest do
       )
     end
   end
+
+  @tag :cuda_required
+  test "regression on single-dimensional and multi-dimensional Random.shuffle" do
+    # these are put in the process dictionary, so it's thread-safe to do this
+    Nx.default_backend({EXLA.Backend, client: :cuda})
+    Nx.Defn.default_options(compiler: EXLA, client: :cuda)
+    key = Nx.Random.key(127)
+
+    t1 = Nx.iota({2, 100})
+    t2 = Nx.iota({100})
```

Comment:
Does the test fail with the previous version? (the tensor is small)

Reply:
Nah. I'm gonna remove this, specially now that we found the actual culprit